### PR TITLE
SDK changes to support new version of swagger Resources API 2020-06-01

### PR DESCRIPTION
**Pull request by Xynoclafe**
SDK changes to support new version of swagger Resources API 2020-06-01

There's some work being done on the service side which has to go in before this is merged, but I'd like to start the review off on this.

### CODE REVIEW

**Comment by isra-fel** [comment]
LGTM. Waiting for service status to be updated.

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/ITagsOperationOperations.cs [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    using Models;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// TagsOperationOperations operations.
+    /// </summary>
+    public partial interface ITagsOperationOperations
```
Comment (filizt) [inline sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/ITagsOperationOperations.cs:24]: >ublic partial interface ITagsOperationOperations [](start = 5, length = 48)

After the changes we reverted on Tags, I'd think the name of the interface would be "ITagsOperations" - could you check on this please?

File: sdk/resources/Microsoft.Azure.Management.Resource/tests/ScenarioTests/TemplateSpecsTests.ScenarioTests.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ResourceGroups.Tests
+
+{
+    class LiveTemplateSpecsTest : TestBase
```
Comment (filizt) [inline sdk/resources/Microsoft.Azure.Management.Resource/tests/ScenarioTests/TemplateSpecsTests.ScenarioTests.cs:9]: >class LiveTemplateSpecsTest : TestBase [](start = 4, length = 38)

We should probably communicate with @StuartKo and add Template Spec tests.

**Comment by filizt** [comment]
>     public class LiveDeploymentTests : TestBase

Do you mind adding a test for the statusMessage update we made?

---
Refers to: sdk/resources/Microsoft.Azure.Management.Resource/tests/ScenarioTests/DeploymentTests.ScenarioTests.cs:18 in fd9e77b. [](commit_id = fd9e77b018afe564b18914c6b655957004a15c7e, deletion_comment = True)

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/ITagsOperationOperations.cs (hunk 1 above)
Comment (Xynoclafe) [reply sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/ITagsOperationOperations.cs:24]: I think this is leftover from the time we generated the SDK before reverting those changes. I don't think it's being used anywhere (I see that ITagsOperations exists separately), I should be able to just remove it. I will look into it.

File: sdk/resources/Microsoft.Azure.Management.Resource/tests/ScenarioTests/TemplateSpecsTests.ScenarioTests.cs (hunk 2 above)
Comment (Xynoclafe) [reply sdk/resources/Microsoft.Azure.Management.Resource/tests/ScenarioTests/TemplateSpecsTests.ScenarioTests.cs:9]: I had created this just as a placeholder of sorts. I've already talked to him about tests.

**Comment by Xynoclafe** [comment]
> > ```
> > public class LiveDeploymentTests : TestBase
> > ```
> 
> Do you mind adding a test for the statusMessage update we made?
> 
> Refers to: sdk/resources/Microsoft.Azure.Management.Resource/tests/ScenarioTests/DeploymentTests.ScenarioTests.cs:18 in [fd9e77b](https://github.com/Azure/azure-sdk-for-net/commit/fd9e77b018afe564b18914c6b655957004a15c7e). [](commit_id = [fd9e77b](https://github.com/Azure/azure-sdk-for-net/commit/fd9e77b018afe564b18914c6b655957004a15c7e), deletion_comment = True)

I added a test for this in InMemoryTests. I was waiting for statusMessage update to be fixed completely before adding a test in LiveDeploymentTests.

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/ITagsOperationOperations.cs (hunk 1 above)
Comment (Xynoclafe) [reply sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/ITagsOperationOperations.cs:24]: Just looked into it and yes, they were leftover from previous SDK generation as the TagsOperations files did not replace them. The class are not referenced anywhere outside their own definitions/extensions and build/tests pass successfully even on removing them.

**Comment by filizt** [comment]
LGTM

**Comment by isra-fel** [comment]
Anything else to add to this PR?
PS there are merging conflicts

**Comment by Xynoclafe** [comment]
> Anything else to add to this PR?
> PS there are merging conflicts

I think we have everything we need. We should be okay to merge this now. I'll push a commit fixing merge issues in a while.

**Comment by Xynoclafe** [comment]
All changes are in.

**Comment by shenglol** [comment]
@isra-fel mind helping merge the PR? We have another minor SDK update for resources that is pending on this.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**